src/twisted/web/wsgi.py

```diff
@@ -455,6 +455,10 @@ def write(self, data):
         # Thing at some point in the future.
 
         def wsgiWrite(started):
+            if self._requestFinished:
+                # Self.request.notifyFinish() has triggered, and
+                # transport is potentially closed.
```

Comment:
Maybe: `Do nothing as this is late write requested.`

but is it ok to just ignore this case?

I was expecting that this might be due to some bad code on the end users, so we should at least log an error message

Reply:
It is not OK, since we don't know (and have no direct control over) when this code will run. Since the wsgi app is run in a thread pool, all data returned must be handed over to the main I/O thread (using callFromThread), and whilst that data is waiting to be written, the transport may be closed (the race in question).